trace_decoder/src/iface.rs

```diff
@@ -0,0 +1,185 @@
+//! Public types for this crate.
+//!
+//! These are all in one place because they're about to be heavily refactored.
+
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+
+use ethereum_types::{Address, U256};
+use evm_arithmetization::proof::{BlockHashes, BlockMetadata};
+use keccak_hash::H256;
+use mpt_trie::partial_trie::HashedPartialTrie;
+use serde::{Deserialize, Serialize};
+
+/// Core payload needed to generate proof for a block.
+/// Additional data retrievable from the blockchain node (using standard ETH RPC
+/// API) may be needed for proof generation.
```

Comment:
the `may` is slightly confusing, we _cannot_ generate proofs without the additional data

Reply:
This is verbatim from the old code, I'll refactor all these structures later